src/schema/v2/fields/__tests__/money.test.ts

```diff
@@ -68,3 +70,165 @@ describe(amount, () => {
     )
   })
 })
+
+describe("major(convertTo:)", () => {
+  const mockArtworkContext = mockArtwork => {
+    return {
+      artworkLoader: () => {
+        return Promise.resolve(mockArtwork)
+      },
+      exchangeRatesLoader: () => {
+        return Promise.resolve({
+          EUR: 5.0,
+        })
+      },
+    }
+  }
+
+  it("converts an exact price from native currency to USD dollars", () => {
+    const context = mockArtworkContext({
+      id: "some-european-artwork",
+      price_currency: "EUR",
+      price_cents: [1000],
+    })
+
+    const query = gql`
+      {
+        artwork(id: "some-european-artwork") {
+          listPrice {
+            ... on Money {
+              original: major
+              converted: major(convertTo: "USD")
+            }
+          }
+        }
+      }
+    `
+
+    return runQuery(query, context).then(result => {
+      expect(result!.artwork.listPrice).toEqual({
+        original: 10.0,
+        converted: 2.0,
+      })
+    })
+  })
+
+  it("converts a price range from native currency to USD dollars", () => {
+    const context = mockArtworkContext({
+      id: "some-european-artwork",
+      price_currency: "EUR",
+      price_cents: [1000, 2000],
+    })
+
+    const query = gql`
+      {
+        artwork(id: "some-european-artwork") {
+          listPrice {
+            ... on PriceRange {
+              minPrice {
+                original: major
+                converted: major(convertTo: "USD")
+              }
+              maxPrice {
+                original: major
+                converted: major(convertTo: "USD")
+              }
+            }
+          }
+        }
+      }
+    `
+
+    return runQuery(query, context).then(result => {
+      expect(result!.artwork.listPrice.minPrice).toEqual({
+        original: 10.0,
+        converted: 2.0,
+      })
+      expect(result!.artwork.listPrice.maxPrice).toEqual({
+        original: 20.0,
+        converted: 4.0,
+      })
+    })
+  })
+
+  it("only supports USD", () => {
+    const context = mockArtworkContext({
+      id: "some-european-artwork",
+      price_currency: "EUR",
+      price_cents: [1000],
+    })
+
+    const query = gql`
+      {
+        artwork(id: "some-european-artwork") {
+          listPrice {
+            ... on Money {
+              major(convertTo: "CAD")
+            }
+          }
+        }
+      }
+    `
+
+    expect(runQuery(query, context)).rejects.toThrowError(/*Only USD*/)
```

Comment:
This throw is not working as expected. It _does_ throw, but not with my supplied error message (which is why it's commented out here). I think maybe there's something about the promise rejection handling I'm not getting.

![throw](https://user-images.githubusercontent.com/140521/70179242-af085900-16ab-11ea-8e66-f2c112e8ec9a.png)

Reply:
Ya, based on that stack trace it looks like a possible issue with the stitching lib, and throwing an error from a union type field. I wouldn't worry too much about it, can timebox and pend the test.

Reply:
I tried but couldn't fix the test, pending for now!